### PR TITLE
[WFLY-18818]:Print log info to let user know the cxf dependenc…

### PR DESCRIPTION
**Pull request by jimma**
https://issues.redhat.com/browse/WFLY-18818

### CODE REVIEW

**Comment by bstansberry** [comment]
@jimma Any logging that was reasonably likely to be unnecessary would need to be INFO or lower, as many users treat WARN as something they have to deal with and get annoyed about anything incorrect.

**Comment by jamezp** [comment]
If I'm understanding correctly a warning might be right here. What his is doing is checking if there is a module dependency on `org.apache.cxf`, `org.apache.cxf.impl` or `org.jboss.ws.cxf.jbossws-cxf-client`. If the dependency is not there and there is an annotation with a package prefix of `org.apache.cxf`, it warns that a module dependency is required for this to work.

I'm not sure we need to log a warning for every annotation, but it does feel a warning or even error might be appropriate. I assume the deployment will not work as expected.

**Comment by jimma** [comment]
> @jimma Any logging that was reasonably likely to be unnecessary would need to be INFO or lower, as many users treat WARN as something they have to deal with and get annoyed about anything incorrect.

@bstansberry This changed to not check the cxf module dependency in parent deployment and sibling deployment because of the ModuleSpecification isn't thread safe. I think this covers most of the dependency declaration. If this deployment processor doesn't find the cxf module dependency in the current deployment, it prints some warning message to ask user checks manually. This warning message is kind of reminder or request for user to check the module dependency is correctly configured. Even we can check the cxf module dependency in parent and sibling deployment after we fix [WFCORE-6639](https://issues.redhat.com/browse/WFCORE-6639), the warning message can still be incorrect when user configure this cxf dependency as a global module.

**Comment by jimma** [comment]
> If I'm understanding correctly a warning might be right here. What his is doing is checking if there is a module dependency on `org.apache.cxf`, `org.apache.cxf.impl` or `org.jboss.ws.cxf.jbossws-cxf-client`. If the dependency is not there and there is an annotation with a package prefix of `org.apache.cxf`, it warns that a module dependency is required for this to work.
> 
That's correct. I only removed the check the module dependency from parent and sibling deployment because of the  ModuleSpecification thread safe issue.  

> I'm not sure we need to log a warning for every annotation, but it does feel a warning or even error might be appropriate. I assume the deployment will not work as expected.
Good point about log a waring message for every annotation @jamezp. This can be improved to print only one warning message as it is only one module dependency is required for this deployment. 
If there isn't cxf module dependency in the current deployment and it is in parent ear or it is in the sibling deployment, it still works, but it beyonds the current deployment processor duty and we can print a warning message to remind user it needs this cxf dependency and user should make sure there is this module dependency whether from parent, sibling or the global module.

File: webservices/server-integration/src/main/java/org/jboss/as/webservices/deployers/WSClassVerificationProcessor.java [hunk 1]
```diff
@@ -99,68 +101,32 @@ void verifyJwsEndpoint(final Class<?> endpointClass, final WebService webService
...
+        final DotName[] dotNames = {WEB_SERVICE_ANNOTATION, WEB_SERVICE_PROVIDER_ANNOTATION};
+        for (final DotName dotName : dotNames) {
+            for (AnnotationInstance ai : index.getAnnotations(dotName)) {
+                AnnotationTarget at = ai.target();
+                if (at instanceof ClassInfo) {
+                    final ClassInfo clazz = (ClassInfo)ai.target();
+                    for (DotName dn : clazz.annotationsMap().keySet()) {
+                        if (dn.toString().startsWith("org.apache.cxf")) {
+                            WSLogger.ROOT_LOGGER.checkModuleDependency(dn.toString(), clazz.name().toString(), "org.apache.cxf");
+                        }
```
Comment (jamezp) [inline webservices/server-integration/src/main/java/org/jboss/as/webservices/deployers/WSClassVerificationProcessor.java:126]: I would just return here after you log.

Comment (jimma) [reply webservices/server-integration/src/main/java/org/jboss/as/webservices/deployers/WSClassVerificationProcessor.java:127]: Done.

**Comment by bstansberry** [comment]
@jimma Leaving this message at WARN will essentially regress the https://issues.redhat.com/browse/WFLY-13832 / https://issues.redhat.com/browse/JBEAP-19682 bug fix.

Changing it to INFO avoids that; otherwise we can go with https://issues.redhat.com/browse/WFCORE-7100 / https://github.com/wildfly/wildfly-core/pull/6285 and bend the rule about how to use ModuleSpecification.

**Comment by jimma** [comment]
@bstansberry @jamezp  Fixed. Please review again. Thanks.

File: webservices/server-integration/src/main/java/org/jboss/as/webservices/logging/WSLogger.java [hunk 2]
```diff
@@ -225,19 +225,9 @@ public interface WSLogger extends BasicLogger {
...
-
     @Message(id = 54, value = "Config %s: %s handler chain with id %s doesn't exist")
     OperationFailedException missingHandlerChain(String configName, String handlerChainType, String handlerChainId);
 
-    // @Message(id = 55, value = "Config %s, %s handler chain %s: doesn't contain handler with name %s")
-    // OperationFailedException missingHandler(String configName, String handlerChainType, String handlerChainId, String handlerName);
-
-    //@LogMessage(level = ERROR)
-    //@Message(id = 56, value = "Method invocation failed with exception: %s")
-    //void methodInvocationFailed(@Cause Throwable cause, String message);
```
Comment (bstansberry) [inline webservices/server-integration/src/main/java/org/jboss/as/webservices/logging/WSLogger.java:239]: Please restore the commented methods. They provide documentation of what the message id was used for and help prevent reuse for other purposes by showing any gaps in the currently used id #s was intentional.

File: webservices/server-integration/src/main/java/org/jboss/as/webservices/logging/WSLogger.java [hunk 3]
```diff
@@ -264,10 +254,6 @@ public interface WSLogger extends BasicLogger {
     @Message(id = 64, value = "Could not update WS server configuration because of existing WS deployment on the server.")
     DisabledOperationException couldNotUpdateServerConfigBecauseOfExistingWSDeployment();
 
-    @LogMessage(level = WARN)
-    @Message(id = 65, value = "Annotation '@%s' found on class '%s'. Perhaps you forgot to add a '%s' module dependency to your deployment?")
-    void missingModuleDependency(String annotation, String clazz, String module);
```
Comment (bstansberry) [inline webservices/server-integration/src/main/java/org/jboss/as/webservices/logging/WSLogger.java:269]: Please restore this but comment it out.

File: webservices/server-integration/src/main/java/org/jboss/as/webservices/logging/WSLogger.java [hunk 4]
```diff
@@ -277,12 +263,6 @@ public interface WSLogger extends BasicLogger {
     @Message(id = 67, value = "Could not activate the webservices subsystem.")
     void couldNotActivateSubsystem(@Cause Throwable cause);
 
-//    @Message(id = 68, value = "Service %s not available")
-//    OperationFailedException serviceNotAvailable(String serviceName);
-
-//    @Message(id = 69, value = "String format password is required")
-//    IllegalArgumentException invalidPasswordType();
```
Comment (bstansberry) [inline webservices/server-integration/src/main/java/org/jboss/as/webservices/logging/WSLogger.java:284]: Please restore.

**Review by bstansberry — CHANGES_REQUESTED** [review]
The main change LGTM; thanks!

I requested some things be restored to the logger file.

**Comment by bstansberry** [comment]
Simple via of TC test status (mostly for my own benefit this today): https://ci.wildfly.org/project.html?projectId=WF_PullRequest&branch_WF_PullRequest=pull%2F17479

**Pull request merged**